unit-tests/py/rspy/devices.py

```diff
@@ -371,16 +385,19 @@ def by_configuration( config, exceptions = None ):
     raised!
     """
     exceptions = exceptions or set()
-    if len( config ) == 1 and re.fullmatch( r'each\(.+\)', config[0], re.IGNORECASE ):
-        spec = config[0][5:-1]
-        for sn in _get_sns_from_spec( spec ):
-            if sn not in exceptions:
-                yield { sn }
+    if re.fullmatch( r'each\(.+\)', config[0], re.IGNORECASE ):
```

Comment:
Do we always have a config? looks like the protection from the out of bound access was removed